tests/commonlib/io_utils.py

```diff
@@ -129,17 +131,62 @@ def exec_command(container_name: str, command: str, param_value: str, resource:
         if command == 'chmod':
             os.chmod(path=resource, mode=int(param_value, base=8))
         elif command == 'chown':
-            uid_gid = param_value.split(':')
-            if len(uid_gid) != 2:
-                raise Exception(
-                    "User and group parameter shall be separated by ':' ")
-            shutil.chown(path=resource, user=uid_gid[0], group=uid_gid[1])
+            try:
+                uid, gid = param_value.split(':')
+            except ValueError as exc:
+                raise Exception("User and group parameter shall be separated by ':' ") from exc
+
+            FsClient.add_users_to_node([uid, gid], in_place=True)
+            shutil.chown(path=resource, user=uid, group=gid)
         elif command == 'unlink':
             if not Path(param_value).is_dir():
                 Path(param_value).unlink()
         else:
             raise Exception(
                 f"Command '{command}' still not implemented in test framework")
+    
+    @staticmethod
+    def add_users_to_node(users: list, in_place: bool):
+        """
+        This function creates the given users along with groups with the
+        same name, on the local container as well the host node.
+        @param users: List of users to create.
+        @param in_place: Whether host node configuration files should be modified in-place or overwritten.
+        @return: None
+        """
+        if in_place:
+            host_users_file = Path('/hostfs/etc/passwd')
+            host_groups_file = Path('/hostfs/etc/group')
+
+            temp_etc = Path('/tmp/etc')
+            temp_etc.mkdir(parents=True, exist_ok=True)
+
+            temp_users_file = temp_etc / 'passwd'
+            temp_groups_file = temp_etc / 'group'
+
+            shutil.copyfile(host_users_file, temp_users_file)
+            shutil.copyfile(host_groups_file, temp_groups_file)
+
+            for user in users:
+                # These commands fail silently for users/groups that exist.
+                subprocess.run(['groupadd', user, '-P', '/tmp'], capture_output=True)
+                subprocess.run(['useradd', user, '-g', user, '-P', '/tmp'], capture_output=True)
+                subprocess.run(['useradd', user], capture_output=True) # For container to get around chmod check.
+
+            FsClient.in_place_copy(temp_users_file, host_users_file)
+            FsClient.in_place_copy(temp_groups_file, host_groups_file)
```

Comment:
@yashtewari why do we need a new method that just copies the content?
shutil has several functions to copy data:
copy()
copy2()
copyfile()
I think we can find appropriate function for our case.

Reply:
All of these methods replace the existing file at the destination, which is undesirable for testing this flow. Check out [this comment](https://github.com/elastic/security-team/issues/4311#issuecomment-1202182095) for more on why we want to test the flow with an in-place copy.